services/bls_aggregation/blsagg.go

```diff
@@ -0,0 +1,364 @@
+// Assumptions:
+//  1. BlsAggregatorService only verifies digest signatures, so avs code needs to verify that the digest
+//     passed to ProcessNewSignature is indeed the digest of a valid taskResponse
+//     (see the comment above checkSignature for more details)
+//  2. BlsAggregatorService is VERY generic and makes very few assumptions about the tasks structure or
+//     the time at which operators will send their signatures. It is mostly suitable for offchain computation
+//     oracle (a la truebit) type of AVS, where tasks are sent onchain by users sporadically, and where
+//     new tasks can start even before the previous ones have finished aggregation.
+//     AVSs like eigenDA that have a much more controlled task submission schedule and where new tasks are
+//     only submitted after the previous one's response has been aggregated and responded onchain, could have
+//     a much simpler AggregationService without all the complicated parallel goroutines.
+type BlsAggregatorService struct {
+	// ResponseC is the channel which all goroutines share to send their responses back to the
+	// main thread after they are done aggregating (either they reached the threshold, or timeout expired)
+	ResponseC chan BlsAggregationServiceResponse
+	// taskChans are the channels to send the signed tasks to the goroutines processing them
+	// each new task is assigned a new goroutine and a new channel
+	taskChans map[types.TaskIndex]chan types.SignedTaskResponseDigest
+	// we add chans to taskChans from the main thread (InitializeNewTask) when we create new tasks,
+	// we read them in ProcessNewSignature from the main thread when we receive new signed tasks,
+	// and remove them from its respective goroutine when the task is completed or reached timeout
+	// we thus need a mutex to protect taskChans
+	taskChansMutex     sync.RWMutex
+	avsRegistryService avsregistry.AvsRegistryService
+	logger             logging.Logger
+}
+
+var _ BlsAggregationService = (*BlsAggregatorService)(nil)
+
+func NewBlsAggregatorService(avsRegistryService avsregistry.AvsRegistryService, logger logging.Logger) *BlsAggregatorService {
+	return &BlsAggregatorService{
+		ResponseC:          make(chan BlsAggregationServiceResponse),
+		taskChans:          make(map[types.TaskIndex]chan types.SignedTaskResponseDigest),
+		taskChansMutex:     sync.RWMutex{},
+		avsRegistryService: avsRegistryService,
+		logger:             logger,
+	}
+}
+
+func (a *BlsAggregatorService) GetResponseChannel() <-chan BlsAggregationServiceResponse {
+	return a.ResponseC
+}
+
+func (a *BlsAggregatorService) InitializeNewTask(
+	taskIndex types.TaskIndex,
+	taskCreatedBlock uint32,
+	quorumNumbers []types.QuorumNum,
+	quorumThresholdPercentages []types.QuorumThresholdPercentage,
+	timeToExpiry time.Duration,
+) error {
+	if _, taskExists := a.taskChans[taskIndex]; taskExists {
+		return TaskAlreadyInitializedErrorF(taskIndex)
+	}
+	taskC := make(chan types.SignedTaskResponseDigest)
```

Comment:
why not name `taskC` as `taskRespC`? It is named as such in `singleTaskAggregatorGoroutineFunc`.

Reply:
Good catch. Updated channel names to aggregatedResponsesC and signedTaskRespsCs: [7172ee2](https://github.com/Layr-Labs/eigensdk-go/pull/24/commits/7172ee23384a8921450b45c8c8c3d123b2ad6422)